### PR TITLE
Rx: resultSelector should be optional (3)

### PR DESCRIPTION
**Pull request by ntrrgc**
Continuation of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9421.

### CODE REVIEW

**Comment by mhegazy** [comment]
@Igorbek can you review this change?

File: rx/rx-lite.d.ts [hunk 1]
```diff
@@ -225,48 +225,30 @@ declare namespace Rx {
...
-		withLatestFrom<T2, T3, T4, TResult>(second: IPromise<T2>, third: IPromise<T3>, fourth: IPromise<T4>, resultSelector?: (v1: T, v2: T2, v3: T3, v4: T4) => TResult): Observable<TResult>;
-		withLatestFrom<T2, T3, T4, T5, TResult>(second: Observable<T2>, third: Observable<T3>, fourth: Observable<T4>, fifth: Observable<T5>, resultSelector?: (v1: T, v2: T2, v3: T3, v4: T4, v5: T5) => TResult): Observable<TResult>;
-		withLatestFrom<TOther, TResult>(souces: Observable<TOther>[], resultSelector?: (firstValue: T, ...otherValues: TOther[]) => TResult): Observable<TResult>;
-		withLatestFrom<TOther, TResult>(souces: IPromise<TOther>[], resultSelector?: (firstValue: T, ...otherValues: TOther[]) => TResult): Observable<TResult>;
-		concat(...sources: Observable<T>[]): Observable<T>;
-		concat(...sources: IPromise<T>[]): Observable<T>;
-		concat(sources: Observable<T>[]): Observable<T>;
-		concat(sources: IPromise<T>[]): Observable<T>;
+		combineLatest<T2>(second: Observable<T2>|IPromise<T2>): Observable<[T, T2]>;
+                combineLatest<T2, TResult>(second: Observable<T2>|IPromise<T2>, resultSelector: (v1: T, v2: T2) => TResult): Observable<TResult>;
```
Comment (Igorbek) [inline rx/rx-lite.d.ts]: Were these indentions intentional? I'm not sure it meets guidelines (if we had them). Could you please align. I'm pedantic, sorry.

File: rx/rx-lite.d.ts [hunk 2]
```diff
@@ -349,15 +323,15 @@ declare namespace Rx {
 		select<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;
 		map<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;	// alias for select
 		pluck<TResult>(prop: string): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => Observable<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => IPromise<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
+		selectMany<TOther, TResult>(selector: (value: T) => Observable<TOther>, resultSelector: (item: T, other: TOther) => TResult): Observable<TResult>;
+		selectMany<TOther, TResult>(selector: (value: T) => IPromise<TOther>, resultSelector: (item: T, other: TOther) => TResult): Observable<TResult>;
```
Comment (Igorbek) [inline rx/rx-lite.d.ts]: Merge these to `selector: (value: T) => Observable<TOther>|IPromise<TOther>`?

File: rx/rx-lite.d.ts [hunk 3]
```diff
@@ -349,15 +323,15 @@ declare namespace Rx {
 		select<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;
 		map<TResult>(selector: (value: T, index: number, source: Observable<T>) => TResult, thisArg?: any): Observable<TResult>;	// alias for select
 		pluck<TResult>(prop: string): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => Observable<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
-		selectMany<TOther, TResult>(selector: (value: T) => IPromise<TOther>, resultSelector?: (item: T, other: TOther) => TResult): Observable<TResult>;
```
Comment (Igorbek) [inline rx/rx-lite.d.ts]: Haven't you added versions where `resultSelector` is omitted?

File: rx/rx-lite.d.ts (hunk 1 above)
Comment (ntrrgc) [inline rx/rx-lite.d.ts]: No, it wasn't.

File: rx/rx-lite.d.ts (hunk 3 above)
Comment (ntrrgc) [inline rx/rx-lite.d.ts]: No, they already existed (check out the lines immediately following)

**Comment by Igorbek** [comment]
@ntrrgc Thanks. Is it work in progress or you want that PR to be merged as is?
You rolled back some optionals without introducing overloads without them. Are you gonna add more or give up with that? :)

File: rx/rx-lite.d.ts (hunk 2 above)
Comment (ntrrgc) [inline rx/rx-lite.d.ts]: heh. There are many declarations that could be merged. I focused in merging only those that caused combinatorial explosion.

File: rx/rx-lite.d.ts (hunk 3 above)
Comment (Igorbek) [inline rx/rx-lite.d.ts]: Ah! My bad

**Comment by ntrrgc** [comment]
It's supposed to be merged as is (minus the indentation issue).

Which optionals did you found without an overload?

**Comment by Igorbek** [comment]
That was in my comment - I was wrong. Ok, Are you gonna fix indention?

**Comment by ntrrgc** [comment]
At this moment I'm not in my main computer and I don't have a clone of DefinitivelyTyped (it would take a while to download). Could you fix it instead, please?

**Comment by Igorbek** [comment]
Thanks. I can't fix your PR. So let's leave it as it for now.

**Pull request merged**